### ISSUE TITLE
Add a "Remember ..." checkbox to the LoginDialog

### ISSUE THREAD

**Issue by manekovskiy** [MahApps/MahApps.Metro#2305]
It would be nice to have a checkbox on `LoginDialog`. From my experience there are a lot of cases when users want to save their username or both username and password.

![checkbox-on-login-dialog](https://cloud.githubusercontent.com/assets/873602/12203721/220bb97e-b639-11e5-9298-9fe6c4d25ff1.png)

The checkbox text and visibility should be configurable in the same way as for example `AffirmativeButton` and `NegativeButton`. And the `LoginDialogData` will be extended with a new flag property indicating whether checkbox is checked or not.

I'm definitely going to implement this feature in my fork so if you are interested I'll send a pull request once it is ready.

**Comment by bigworld12** [comment]
i agree this should be added

**Comment by xxMUROxx** [comment]
Agree :+1:

But then we have to store some information into the application settings. Each *logon dialog* must have an ID that identifies the dialogue, whether it should be opened or not---depending on the selection.

**Comment by manekovskiy** [comment]
@xxMUROxx When we need to access anything that requires authentication we typically have some abstractions like "Authenticator" and "CredentialsStore". Authenticator can either ask CredentialsStore for saved credentials or start the authentication process from the very beginning which involves collecting a username/password using the LoginDialog.
And CredentialsStore knows how to save/retrieve credentials, it is a way to abstract out the place where credentials live between user sessions.

When I'm trying to imagine interactions between this Authenticator/CredentialsStore pair I cannot see "credentials identifier" as a part of LoginDialog, it would be more practical to have it as a part of Authenticator implementation since he is the guy who talks with CredentialsStore.

**Comment by thoemmi** [comment]
@xxMUROxx why should the settings be stored by MahApps.Metro? Username and password aren't stored either. It's up to the application (or business logic respectively) to both provide the login dialog with a `LoginDialogSettings` instance and to process its returned `LoginDialogData`. MahApps.Metro is meant for the presentation layer only.
